posts/comparing-java-lts-releases.md

```diff
@@ -0,0 +1,128 @@
+
+Another component to Java 9 that is worth mentioning is the introduction of modules.  Modules introduce a re-organization of Java APIs, by grouping common libraries and specifying groups of dependent libraries.  These groups create modules, and they define an entirely new way to build applications.  Using the `javac` compiler, you can specify a `module-info.java` along with any relevant class files.  What results is a module that you can run in the command line via the `java` command.  You no longer need to compile JAR files or memorize the different program entry points you have in your JAR or your manifest file.  For example, given the class `HelloWorld` belonging to package `com.mydomain`, you can write a `module-info.java` file containing the line `module com.mydomain {}`.  Then with the following command, you can compile a java module
+```bash
+javac -d mymodule/myproject src/myproject/module-info.java src/myproject/com/mydomain/HelloWorld.java
+java --module-path mymodule -m myproject/com.mydomain.HelloWorld
+```
+Modules belong to a wider initiative known as Project Jigsaw, which again extends beyond the scope of this blog post.  For information on modules, how they work, and Project Jigsaw, [check out this article](http://openjdk.java.net/projects/jigsaw/quick-start)
+
+# Java 10 Non-LTS
+Java 9 was the release that re-defined how Java programs could be organized and developed.  Java 10 was the release that built upon the foundation laid in Java 9.  The most subtle, but most important feature of this release were the performance enhancements introduced to the jshell tool.  The jshell tool was a powerful addition in Java 9, but it took awhile for it to get going.  Java 10 kicked it into gear a lot faster than Java 9 did, introducing performance enhancements that made the tool more desirable to use for developers.  
+
+Additionally, Java 10 improved upon Java 9's new Garbage Collector, the Garbage-First Garbage Collector.  Java 9's G1GC changed the way Java performed garbage collecting by partitioning its heap into multiple small heaps, whereas traditionally the heap would be split into three chunks.  G1GC also introduced multi-threaded processing on the garbage collector for all events save a "stop-the world," a point in time during garbage collection where program execution halts and gives the garbage collector time to assess the state of the application.  "Stop-the-world" events are unaviodable in garbage collection.  Java 10 took the multi-threaded approach perfected by Java 9 and applied it to the "stop-the-world" event, thus making the G1GC fully concurrent.  Even though the G1GC is fully concurrent and therefore faster than prior garbage collectors on average, it is worth noting it takes a larger memory footprint because of the way it chops the heap into more chunks.
```

Comment:
```suggestion
Additionally, Java 10 improved upon Java 9's new Garbage Collector, the Garbage-First Garbage Collector.  Java 9's G1GC changed the way Java performed garbage collecting by partitioning its heap into multiple small heaps, whereas traditionally the heap would be split into three chunks.  G1GC also introduced multi-threaded processing on the garbage collector for all events save a "stop-the world," a point in time during garbage collection where program execution halts and gives the garbage collector time to assess the state of the application.  "Stop-the-world" events are unavoidable in garbage collection.  Java 10 took the multi-threaded approach perfected by Java 9 and applied it to the "stop-the-world" event, thus making the G1GC fully concurrent.  Even though the G1GC is fully concurrent and therefore faster than prior garbage collectors on average, it is worth noting it takes a larger memory footprint because of the way it chops the heap into more chunks.
```